src/SixtyNine/Cloud/Factory/FontsFactory.php

```diff
@@ -13,38 +13,38 @@
 
 class FontsFactory
 {
-    /** @var string */
-    protected $fontsPath;
     /** @var array */
     protected $fonts = array();
 
     /**
-     * @param string $fontsPath
+     * @param string|array $fontsPath
      */
     protected function __construct($fontsPath)
     {
-        $this->fontsPath = $fontsPath;
-        $this->loadFonts();
+        if (is_array($fontsPath)) {
+            foreach ($fontsPath as $filename) {
+                Assert::fileExists($filename, 'The fonts path must exist');
+                $name = basename($filename);
+                $this->fonts[$name] = new Font($name, realpath($filename));
+            }
+        } else {
+            Assert::fileExists($fontsPath, 'The fonts path must exist');
+            foreach (glob($fontsPath . "/*.ttf") as $filename) {
+                $name = basename($filename);
+                $this->fonts[$name] = new Font($name, realpath($filename));
+            }
+        }
     }
 
     /**
      * @param string $fontsPath
      * @return FontsFactory
      */
-    public  static function create($fontsPath)
+    public static function create($fontsPath)
     {
-        Assert::fileExists($fontsPath, 'The fonts path must exist');
         return new self($fontsPath);
     }
 
-    public function loadFonts()
```

Comment:
Up to you, but I would personnaly keep this function with the business you've put in the constructor and still call it from the constructor (just for the structure). This can be turned to a protected function when it makes no sense to call it from outside.